objects/src/transaction/proven_tx.rs

```diff
@@ -159,6 +125,292 @@ impl ProvenTransaction {
     }
 }
 
+// PROVEN TRANSACTION BUILDER
+// ================================================================================================
+
+/// Builder for a proven transaction.
+#[derive(Clone, Debug)]
+pub struct ProvenTransactionBuilder {
+    /// ID of the account that the transaction was executed against.
+    account_id: Option<AccountId>,
+
+    /// The hash of the account before the transaction was executed.
+    initial_account_hash: Option<Digest>,
+
+    /// The hash of the account after the transaction was executed.
+    final_account_hash: Option<Digest>,
+
+    /// State changes to the account due to the transaction.
+    account_details: Option<AccountDetails>,
+
+    /// List of [Nullifier]s of all consumed notes by the transaction.
+    input_notes: Vec<Nullifier>,
+
+    /// List of [NoteEnvelope]s of all notes created by the transaction.
+    output_notes: Vec<NoteEnvelope>,
+
+    /// State of the output notes.
+    output_note_details: BTreeMap<NoteId, Note>,
+
+    /// The script root of the transaction, if one was used.
+    tx_script_root: Option<Digest>,
```

Comment:
Technically, we don't need `tx_script_root` in a proven transaction. This would require a small kernel change though. Let's create an issue for this.

Reply:
https://github.com/0xPolygonMiden/miden-base/issues/505